### PR TITLE
API support for GroupProjectBinding resources [part 1]

### LINKED ISSUE #10003: GroupProjectBinding API endpoints

### PR DESCRIPTION
**Pull request by pkprzekwas**
**What does this PR do / Why do we need it**:
Introduces `list` and `get` endpoints for GroupProjectBinding resources.

**Does this PR close any issues?**:
Fixes #10003 

**Special notes for your reviewer**:
- Support for `create`/`patch`/`delete` endpoints will be added in separate PRs.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/ee/group-project-binding/handler/handler.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	"k8c.io/kubermatic/v2/pkg/handler/v1/common"
+	"k8c.io/kubermatic/v2/pkg/provider"
+	utilerrors "k8c.io/kubermatic/v2/pkg/util/errors"
+)
+
+// swagger:parameters getGroupProjectBinding
+type getGroupProjectBindingReq struct {
+	// in: path
+	// required: true
+	ProjectID string `json:"project_id"`
```
Comment (lsviben) [inline pkg/ee/group-project-binding/handler/handler.go]: NIT: you can embed ProjectReq for this if you would like

File: pkg/ee/group-project-binding/provider/provider.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+	if err := p.clientPrivileged.List(ctx, allBindings, listOpts); err != nil {
+		return nil, err
+	}
+
+	var projectBindings []kubermaticv1.GroupProjectBinding
+	for _, binding := range allBindings.Items {
+		if binding.Spec.ProjectID == projectID {
+			projectBindings = append(projectBindings, binding)
+		}
+	}
```
Comment (lsviben) [inline pkg/ee/group-project-binding/provider/provider.go]: I plan to add a mutating webhook for adding projectID and ownerRef so lets keep in mind that we can simplify this later

File: pkg/ee/group-project-binding/provider/provider.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+	if len(projectBindings) > 0 {
+		// Fetch first binding to ensure user has permissions
+		_, err := p.Get(ctx, userInfo, projectBindings[0].Name)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		return nil, nil
+	}
```
Comment (lsviben) [inline pkg/ee/group-project-binding/provider/provider.go]: We can also make a better check for these when the group support for KKP api gets merged. Just adding the comment so we keep it in mind

File: pkg/handler/v2/routes_v2.go [hunk 4]
```diff
@@ -6787,3 +6797,61 @@ func (r Routing) deleteResourceQuota() http.Handler {
 		r.defaultServerOptions()...,
 	)
 }
+
+//swagger:route get /api/v2/projects/{project_id}/groupbindings groupProjectBinding listGroupProjectBinding
```
Comment (lsviben) [inline pkg/handler/v2/routes_v2.go]: maybe keep this under project tag?
```suggestion
//swagger:route get /api/v2/projects/{project_id}/groupbindings project listGroupProjectBinding
```

File: pkg/ee/group-project-binding/provider/provider.go (hunk 2 above)
Comment (pkprzekwas) [reply pkg/ee/group-project-binding/provider/provider.go]: What to you think about leaving a comment here keep that in mind?

Comment (lsviben) [reply pkg/ee/group-project-binding/provider/provider.go]: good idea

**Pull request merged**